clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy

```diff
@@ -534,4 +549,21 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
       null
     }
   }
+
+  Map getUserData(BasicGoogleDeployDescription description, String serverGroupName,
+                  String instanceTemplateName, GoogleNamedAccountCredentials credentials) {
+    String customUserData = ''
+    if (description.userDataType && description.userData) {
+      if (description.userDataType.equals("URL")) {
+        task.updateStatus BASE_PHASE, "Resolving user data from url $description.userData..."
+        customUserData = description.userData.toURL()?.text
+      } else {
+        customUserData = description.userData
+      }
+    }
+    Map userData = googleUserDataProvider.getUserData(serverGroupName, instanceTemplateName,
```

Comment:
Custom user data is still applied when the type is "URL", but I can change it so it reads the `description.userData` field as custom user data by default, and reads from a URL only when the type is set as "URL" (so `description.userDataType` won't be necessary if `description.userData` isn't a URL).